src/main/java/org/wildfly/security/auth/util/GSSCredentialSecurityFactory.java

```diff
@@ -322,9 +323,16 @@ private GSSKerberosCredential createGSSCredential(Configuration configuration) t
             final Subject subject = new Subject();
 
             try {
-                final LoginContext lc = new LoginContext("KDC", subject, (c) -> {
-                    throw new FastUnsupportedCallbackException(c[0]);
-                }, configuration);
+                final ClassLoader oldCl = WildFlySecurityManager.getCurrentContextClassLoaderPrivileged();
+                WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(Builder.class.getClassLoader());
+                final LoginContext lc;
+                try {
+                    lc = new LoginContext("KDC", subject, (c) -> {
+                        throw new FastUnsupportedCallbackException(c[0]);
+                    }, configuration);
+                } finally {
+                    WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(oldCl);
```

Comment:
Shouldn't be origin classloader returned after lc.login()? Seems login module is instantiated over there.

Reply:
The context class loader is cached in the ``init`` method which is called from the constructor.

Reply:
right